File: examples/cpp/galbot_robot/src/logger_example.cpp
#include <iostream>

#include "galbot_robot.hpp"
#include "galbot_sdk_logger.hpp"

using namespace galbot::sdk::g1;

int main() {
  // 获取 GalbotRobot 的单例并初始化
  auto& robot = GalbotRobot::get_instance();
  robot.init();
  // ==============================
  // 1. 初始化 SDK 日志
  // ==============================
  LoggerConfig logger_config;
  // 日志生成路径，如不填充，默认生成在~/galbot_sdk_log/user_log路径
  logger_config.path = "";
  // 日志文件名，如不填充，默认日志名称为<process_name>_<current_time>_<pid>_<thread_id>.log
  logger_config.file_name = "";
  // 单个日志文件最大字节数
  logger_config.file_max_size = 1024 * 1024 * 10; // 10MB
  // 循环日志文件最大数量，单个文件写满后新建文件，最多保留 file_max_num 个文件
  logger_config.file_max_num = 5;
  // 输出日志最低等级
  logger_config.level = LogLevel::DEBUG;
  // 是否输出到终端，默认为 false
  logger_config.console_output = false;

  if (!init_galbot_sdk_logger(logger_config)) {
    std::cerr << "failed to init galbot sdk logger" << std::endl;
    return -1;
  } else {
    std::cout << "galbot sdk logger initialized successfully" << std::endl;
  }

  // ==============================
  // 2. 打印不同级别日志
  // ==============================
  GALBOT_SDK_LOG_TRACE << "this is trace log";
  GALBOT_SDK_LOG_DEBUG << "this is debug log";
  GALBOT_SDK_LOG_INFO << "this is info log";
  GALBOT_SDK_LOG_WARN << "this is warning log";
  GALBOT_SDK_LOG_ERROR << "this is error log";
  GALBOT_SDK_LOG_CRITICAL << "this is critical log";

  // ==============================
  // 3. 支持链式 << 风格
  // ==============================
  int robot_id = 3;
  GALBOT_SDK_LOG_INFO << "robot_id = " << robot_id;

  GALBOT_SDK_LOG_INFO << "example program finished";

  // 退出系统并进行SDK资源、logger资源释放
  robot.request_shutdown();
  robot.wait_for_shutdown();
  robot.destroy();
  return 0;
}

File: examples/python/galbot_robot/logger_example.py
import galbot_sdk

cfg = {
    # 日志存放目录，如果为空，则默认生成在 ~/galbot_sdk_log/user_log
    "path": "",

    # 日志文件名，如果为空，则默认名称为 <process_name>_<current_time>_<pid>_<thread_id>.log
    "file_name": "",

    # 单个日志文件最大字节数，如果日志超过此大小，会切换到新文件
    "file_max_size": 10 * 1024 * 1024,  # 10MB

    # 循环日志文件最大数量，超过数量时会覆盖最旧的日志文件
    "file_max_num": 5,

    # 是否在控制台输出日志，默认 False
    "console_output": True,

    # 日志输出等级，可用值：debug, info, warning, error, critical
    "level": "info",
}

galbot_sdk.init_logger(cfg)

# 写日志
galbot_sdk.debug("Debug example")
galbot_sdk.info("Info example")
galbot_sdk.warning("Warning example")
galbot_sdk.error("Error example")
galbot_sdk.critical("Critical example")
